test/integration/modules/bzz.spec.ts

```diff
@@ -1,15 +1,131 @@
 
 const BEE_KY_OPTIONS = beeKyOptions()
 
 describe('modules/bzz', () => {
+  describe('act', () => {
+    const data = 'hello act'
+    let publicKey: string
+    let batchID: BatchId
+    const grantees = [
+      '02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e8',
+      '02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e9',
+      '02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12ee',
+    ]
+    const patchGrantees = {
+      add: ['02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e8'],
+      revoke: [
+        '02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e9',
+        '02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12ee',
+      ],
+    }
+
+    const patchGranteesString = JSON.stringify(patchGrantees)
+
+    before(async () => {
+      const responsePUBK = await http<{ publicKey: string }>(BEE_KY_OPTIONS, {
+        method: 'get',
+        url: 'addresses',
+        responseType: 'json',
+      })
+      publicKey = responsePUBK.data.publicKey
+
+      const responseBATCHID = await http<{ batchID: BatchId }>(BEE_KY_OPTIONS, {
+        method: 'post',
+        url: 'stamps/420000000/17',
+        responseType: 'json',
+      })
+      batchID = responseBATCHID.data.batchID
+    })
+
+    it('should upload with act', async function () {
+      const result = await bzz.uploadFile(BEE_KY_OPTIONS, data, batchID, 'act-1.txt', { act: true })
+      expect(result.reference).to.have.lengthOf(64)
+      expect(result.history_address).to.have.lengthOf(64)
+    })
+
+    it('should not be able to download without ACT header', async function () {
+      const result = await bzz.uploadFile(BEE_KY_OPTIONS, data, batchID, 'act-1.txt', { act: true })
+      await expect(bzz.downloadFile(BEE_KY_OPTIONS, result.reference)).to.be.rejectedWith(
+        'Request failed with status code 404',
+      )
+    })
+
+    it('should not be able to download with ACT header but with wrong publicKey', async function () {
+      const result = await bzz.uploadFile(BEE_KY_OPTIONS, data, batchID, 'act-2.txt', { act: true })
+      const requestOptionsBad = actBeeKyOptions(
+        '0x1234567890123456789012345678901234567890123456789012345678901234',
+        result.history_address,
+        '1',
+      )
+      await expect(bzz.downloadFile(requestOptionsBad, result.reference)).rejectedWith(
+        'Request failed with status code 400',
+      )
+    })
+
+    it('should download with ACT and valid publicKey', async function () {
+      const filename = 'act-3.txt'
+      const result = await bzz.uploadFile(BEE_KY_OPTIONS, data, batchID, filename, { act: true })
+      const requestOptionsOK = actBeeKyOptions(publicKey, result.history_address, '1')
+      const dFile = await bzz.downloadFile(requestOptionsOK, result.reference, filename)
+      expect(Buffer.from(dFile.data).toString()).to.eql(data)
+    })
+
+    it('should create grantee list', async function () {
```

Comment:
the grantee tests should go under a separate 'grantee' test file because it is a separate endpoint

Reply:
New module `grantee` created. https://github.com/Solar-Punk-Ltd/bee-js/commit/e8b35e07b877fe89c61ff2f6a77b238c25f950ea